src/common/mfu_util.c

```diff
@@ -1105,6 +1109,284 @@ bool mfu_is_lustre(const char* path)
+
+        /*
+         * Parse one of the lines in /proc/self/mounts
+         */
+        mfu_mount_info_parse_mount_line(line, len, itemp);
+
+        /*
+         *  Get the device for the mountpoint in question.
+         */
+        errno = 0;
+        rc = stat(itemp->mfu_mi_mountpoint, &st);
+        if (rc == -1) {
+            fclose(mountfp);
+            fprintf(stderr, "Could not stat: %s (%d)", itemp->mfu_mi_mountpoint, errno);
+            return(1);
+        }
+        itemp->mfu_mi_dev = st.st_dev;
+
+        /*
+         *  Now we can insert the item into the list.
+         */
+        list_add_tail(&itemp->mfu_mi_linkage, mfu_mi_list);
+        bytes = getline(&line, &len, mountfp);
+    }
+
+    fclose(mountfp);
+    return(0);
+}
+
+void mfu_init_filesystem_list(void)
+{
+    struct mfu_mount_info *itemp = NULL;
+    struct mfu_mount_info *tmpitemp = NULL;
+    int                    rc;
+
+    if (mfu_mi_list_initialized == true) {
+        MFU_LOG(MFU_LOG_INFO, "The filesystem list has already been initialized...");
+        return;
+    }
+
+    MFU_LOG(MFU_LOG_INFO, "Initializing filesystem list...");
+    INIT_LIST_HEAD(&mfu_mi_list);
+    mfu_mi_list_initialized = true;
+
+    rc = mfu_mount_info_get_filesystem_list(&mfu_mi_list);
+    if (rc != 0) {
+        list_for_each_entry_safe(itemp,
+                                 tmpitemp,
+                                 &mfu_mi_list,
+                                 mfu_mi_linkage) {
+            list_del_init(&itemp->mfu_mi_linkage);
+            mfu_mount_info_destroy(itemp);
+            if ( itemp != NULL ) {
+                free(itemp);
+                itemp = NULL;
+            }
+        }
+
+        return;
+    }
+
+    mfu_mi_list_initialized = true;
+}
+
+void mfu_destroy_filesystem_list(void)
+{
+    struct mfu_mount_info *itemp = NULL;
+    struct mfu_mount_info *tmpitemp = NULL;
+
+    if (mfu_mi_list_initialized == false) {
+        MFU_LOG(MFU_LOG_INFO, "The filesystem list has not been initialized...");
+        return;
+    }
+
+    list_for_each_entry_safe(itemp,
+                             tmpitemp,
+                             &mfu_mi_list,
+                             mfu_mi_linkage) {
+        list_del_init(&itemp->mfu_mi_linkage);
+        mfu_mount_info_destroy(itemp);
+        if ( itemp != NULL ) {
+            free(itemp);
+            itemp = NULL;
+        }
+    }
+}
+
+bool mfu_is_hpss(const char* path)
+{
+    bool is_hpss = false;
+
+#ifdef HPSS_SUPPORT
+    struct mfu_mount_info *itemp = NULL;
+    struct mfu_mount_info *tmpitemp = NULL;
+    struct stat            st;
+    int                    rc;
+
+    if (mfu_mi_list_initialized == false) {
+        MFU_LOG(MFU_LOG_INFO, "The filesystem list has not been initialized...");
+        return(false);
+    }
+
+    errno = 0;
+    rc = stat(path, &st);
+    if (rc == -1) {
+        fprintf(stderr, "Could not stat: %s (%d)", path, errno);
```

Comment:
Let's use ``MFU_LOG(MFU_LOG_ERR, ...`` here.

Reply:
Sounds good. I missed this!